### PR TITLE
feat: add openapi spec for resend modules

### PR DESCRIPTION
**Pull request by kewynakshlley**
* added openapi spec for emails
* added openapi spec for domains
* added openapi spec for api keys

### CODE REVIEW

File: resend.yaml [hunk 1]
```diff
@@ -1,93 +1,548 @@
...
+  description: 'Resend is the email platform for developers.'
 servers:
   - url: https://api.resend.com
-tags:
-  - name: email
-    description: Email operations
 security:
   - bearerAuth: []
+tags:
+  - name: Email
```
Comment (zenorocha) [inline resend.yaml]: Email → Emails

File: resend.yaml [hunk 2]
```diff
@@ -1,93 +1,548 @@
...
+      summary: Trigger up to 100 batch emails at once.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/SendEmailRequest'
+      responses:
+        '200':
```
Comment (zenorocha) [inline resend.yaml:67]: I believe we return 201 instead of 200 for "create" endpoints

File: resend.yaml [hunk 3]
```diff
@@ -1,93 +1,548 @@
...
+          description: Reply-to email address. For multiple addresses, send as an array of strings.
         html:
           type: string
           description: The HTML version of the message.
         text:
           type: string
           description: The plain text version of the message.
+        react:
+          type: string
+          description: The React component used to write the message. Only available in the Node.js SDK.
```
Comment (zenorocha) [inline resend.yaml]: React only works in the Node.js SDK. Can you remove?

File: resend.yaml [hunk 4]
```diff
@@ -1,93 +1,548 @@
...
+          type: string
+          description: The status of the domain.
+        records:
+          type: array
+          items:
+            $ref: '#/components/schemas/DomainRecord'
+        region:
+          type: string
+          description: The region where the domain is hosted.
+        dnsProvider:
```
Comment (zenorocha) [inline resend.yaml]: I believe `dnsProvider` is outdated, can you double check?

**Review by zenorocha — CHANGES_REQUESTED** [review]

File: resend.yaml [hunk 5]
```diff
@@ -1,93 +1,548 @@
...
-tags:
-  - name: email
-    description: Email operations
 security:
   - bearerAuth: []
+tags:
+  - name: Email
+    description: Start sending emails through the Resend Email API.
+  - name: Domains
+    description: Create and manage domains through the Resend Email API.
```
Comment (zenorocha) [inline resend.yaml]: `Resend Email API` → `Resend API`

File: resend.yaml [hunk 6]
```diff
@@ -1,93 +1,548 @@
...
+  - name: API Keys
+    description: Create and manage API Keys through the Resend Email API.
```
Comment (zenorocha) [inline resend.yaml]: `Resend Email API` → `Resend API`

File: resend.yaml [hunk 7]
```diff
@@ -1,93 +1,548 @@
...
 paths:
-  /email:
+  /emails:
     post:
       tags:
-        - email
-      summary: Send an email
-      operationId: sendEmail
+        - Email
+      summary: Sends an email
```
Comment (zenorocha) [inline resend.yaml]: `Sends an email` → `Send an email`

File: resend.yaml [hunk 8]
```diff
@@ -1,93 +1,548 @@
...
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateBatchEmailsResponse'
+  /domains:
+    post:
+      tags:
+        - Domains
+      summary: Creates a new domain.
```
Comment (zenorocha) [inline resend.yaml]: `Creates` → `Create`

File: resend.yaml [hunk 9]
```diff
@@ -1,93 +1,548 @@
...
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/VerifyDomainResponse'
+  /api-keys:
+    post:
+      tags:
+        - API Keys
+      summary: Creates a new API key.
```
Comment (zenorocha) [inline resend.yaml]: `Creates` → `Create`

File: resend.yaml [hunk 10]
```diff
@@ -1,93 +1,548 @@
...
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateDomainResponse'
+    get:
+      tags:
+        - Domains
+      summary: Retrieve a list of domains for the authenticated user.
```
Comment (zenorocha) [inline resend.yaml]: Remove "for the authenticated user"

File: resend.yaml [hunk 11]
```diff
@@ -1,93 +1,548 @@
...
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListDomainsResponse'
+  /domains/{domain_id}:
+    get:
+      tags:
+        - Domains
+      summary: Retrieve a single domain for the authenticated user.
```
Comment (zenorocha) [inline resend.yaml]: Remove "for the authenticated user"

File: resend.yaml [hunk 12]
```diff
@@ -1,93 +1,548 @@
...
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateApiKeyResponse'
+    get:
+      tags:
+        - API Keys
+      summary: Retrieve a list of API keys for the authenticated user.
```
Comment (zenorocha) [inline resend.yaml]: Remove "for the authenticated user"

File: resend.yaml [hunk 13]
```diff
@@ -1,93 +1,548 @@
...
+tags:
+  - name: Email
+    description: Start sending emails through the Resend Email API.
+  - name: Domains
+    description: Create and manage domains through the Resend Email API.
+  - name: API Keys
+    description: Create and manage API Keys through the Resend Email API.
 paths:
-  /email:
+  /emails:
```
Comment (zenorocha) [inline resend.yaml:18]: Can you move this endpoint to be the first one since that's what people use the most?

File: resend.yaml [hunk 14]
```diff
@@ -1,93 +1,548 @@
...
           description: OK
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/SendEmailResponse'
+  /emails/{email_id}:
+    get:
+      tags:
+        - Email
+      summary: Retrieve a single email.
```
Comment (zenorocha) [inline resend.yaml]: Can you remove the ending period (`.`) from all `summary`?

File: resend.yaml (hunk 2 above)
Comment (kewynakshlley) [reply resend.yaml:67]: I did a double check and I can confirm that Send Email and Send Batch returns 200.

File: resend.yaml (hunk 4 above)
Comment (kewynakshlley) [reply resend.yaml]: I've confirmed that dnsProvider doesn't exist on the Node SDK.

File: resend.yaml (hunk 13 above)
Comment (kewynakshlley) [reply resend.yaml:18]: This endpoint is already the first one.  Maybe it's something related to the postman itself?

**Pull request merged**